src/transformers/models/vision_encoder_decoder/modeling_flax_vision_encoder_decoder.py

```diff
@@ -0,0 +1,845 @@
+            also be used by default.
+        decoder_position_ids (:obj:`jnp.ndarray` of shape :obj:`(batch_size, sequence_length)`, `optional`):
+            Indices of positions of each decoder input sequence tokens in the position embeddings. Selected in the
+            range ``[0, config.decoder.max_position_embeddings - 1]``.
+        past_key_values (:obj:`Dict[str, jnp.ndarray]`, `optional`, returned by ``init_cache`` or when passing previous ``past_key_values``):
+            Dictionary of pre-computed hidden-states (key and values in the attention blocks) that can be used for fast
+            auto-regressive decoding. Pre-computed key and value hidden-states are of shape `[batch_size, max_length]`.
+        output_attentions (:obj:`bool`, `optional`):
+            Whether or not to return the attentions tensors of all attention layers. See ``attentions`` under returned
+            tensors for more detail.
+        output_hidden_states (:obj:`bool`, `optional`):
+            Whether or not to return the hidden states of all layers. See ``hidden_states`` under returned tensors for
+            more detail.
+        return_dict (:obj:`bool`, `optional`):
+            If set to ``True``, the model will return a
+            :class:`~transformers.file_utils.FlaxCausalLMOutputWithCrossAttentions` instead of a plain tuple.
+"""
+
+
+class FlaxVisionEncoderDecoderModule(nn.Module):
+    config: VisionEncoderDecoderConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        encoder_config = self.config.encoder
+        decoder_config = self.config.decoder
+
+        # Copied from `modeling_hybrid_clip.py` with modifications.
+        from ...models.auto.modeling_flax_auto import FLAX_MODEL_FOR_CAUSAL_LM_MAPPING, FLAX_MODEL_MAPPING
+
+        encoder_module = FLAX_MODEL_MAPPING[encoder_config.__class__].module_class
+        decoder_module = FLAX_MODEL_FOR_CAUSAL_LM_MAPPING[decoder_config.__class__].module_class
+
+        self.encoder = encoder_module(encoder_config, dtype=self.dtype)
+        self.decoder = decoder_module(decoder_config, dtype=self.dtype)
+
+        # encoder outputs might need to be projected to different dimension for decoder
+        if (
+            self.encoder.config.hidden_size != self.decoder.config.hidden_size
+            and self.decoder.config.cross_attention_hidden_size is None
+        ):
+            self.enc_to_dec_proj = nn.Dense(
+                self.decoder.config.hidden_size,
+                kernel_init=jax.nn.initializers.normal(self.decoder.config.initializer_range, self.dtype),
+                dtype=self.dtype,
+            )
+        else:
+            self.enc_to_dec_proj = None
+
+    def _get_encoder_module(self):
+        return self.encoder
+
+    def _get_projection_module(self):
+        return self.enc_to_dec_proj
+
+    def _get_decoder_module(self):
+        return self.decoder
+
+    def __call__(
+        self,
+        pixel_values,
+        attention_mask,
```

Comment:
`ViT` model does not use `attention_mask`, so IMO we should not add it here. This is also a bit confusing since we pass this `attention_mask` as `encoder_attention_mask` to the decoder, but the encoder does not really use this mask. If a user passes `attention_mask` he would expect it to be applied in the encoder, as this is how it's implemented in all other models.

cc @patrickvonplaten

Reply:
Hey, @patil-suraj 

I am OK to remove `attention_mask`, especially because `VisionEncoderDecoder.forward` doesn't have it (@NielsRogge set it to `None` inside forward pass).

However, I saw `beit` has `bool_masked_pos`, which I think plays a role of `attention_mask`. I have a feeling that the usage of attention mask in future vision models might become more popular, and at some point a change will be required. cc @NielsRogge  to see if he has some opinion.

Reply:
I'm not sure if adding `attention_mask` makes sense for vision models. Models like ViT, BEiT operate on patches of an image, so `attention_mask` would then be used to not include certain patches in the `attention_scores` calculation. Is there a use case for that?

Also, the `bool_masked_pos` indicates which patch tokens to mask in order to do masked image modeling. This is different from attention_mask, which indicates which tokens do not include in the calculation of the attention scores matrix.

Reply:
And yes, currently is set `attention_mask=None` to the vision models, in order for them to work with the generate method. We are discussing internally how to fix this, because it's an ugly hack.

Reply:
> 
> 
> I'm not sure if adding `attention_mask` makes sense for vision models. Models like ViT, BEiT operate on patches of an image, so `attention_mask` would then be used to not include certain patches in the `attention_scores` calculation. Is there a use case for that?
> 

No, as far as I know.

> Also, the `bool_masked_pos` indicates which patch tokens to mask in order to do masked image modeling. This is different from `attention_mask`, which indicates which tokens do not include in the calculation of the attention scores matrix.

OK, thank you for the information!